<?xml version="1.0"?>
<launch>
  <rosparam param="/use_sim_time">True</rosparam>
  <arg name="gui" default="true" doc="Starts sapien gui"/>
  <arg name="paused" default="false" doc="Starts sapien in paused mode"/>
  <arg name="scene" default="1-1" doc="Scene id."/>
  <arg name="materials_dir" default="/root/ocrtoc_materials"  doc="The models and scenes path" />
  <arg name="world_name" value="$(arg materials_dir)/scenes/$(arg scene)/input.world"/>

  <!-- startup simulated world -->
  <arg name="use_gui" value="--gui" if="$(eval gui)"/>
  <arg name="use_gui" value="" unless="$(eval gui)"/>
  <arg name="use_pause" value="--paused" if="$(eval paused)"/>
  <arg name="use_pause" value="" unless="$(eval paused)"/>
  <node pkg="sapien_simulator" type="sapien_env.py" name="sapien_env" respawn="false" output="screen"
        args="$(arg use_gui) $(arg use_pause) --world_name $(arg world_name)">
    <param name="materials_dir" value="$(arg materials_dir)"/>
  </node>

  <!-- load robot definition -->
  <include file="$(find sapien_simulator)/launch/ur5e_upload.launch"/>

  <!-- load controllers -->
  <include file="$(find ocrtoc_solution)/launch/controllers.launch"/>

  <!-- load camera info pub -->
  <include file="$(find sapien_simulator)/launch/camera_info_pub.launch"/>
</launch>
